src/main/java/Duke.java

```diff
@@ -1,10 +1,37 @@
+import duke.Storage;
+import duke.task.TaskList;
+import duke.Ui;
+import duke.Parser;
+import duke.command.Command;
+
 public class Duke {
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+    
+    public Duke(String filePath) {
+        taskList = new TaskList();
+        storage = new Storage(filePath);
+        ui = new Ui(taskList, storage);
+    }
+    
+    public void run() {
+        ui.showWelcome();
+        ui.showLine();
+        boolean isExit = false;
+        while (!isExit) {
+            String command = ui.command();
+            Parser parser = new Parser(command);
+            if (parser.isExit()) {
+                break;
+            }
+            Command c = parser.parse();
```

Comment:
You can consider making `parse()` a static method, that way you would not have to create a new parser object everytime a command is given.